src/dos/dos_classes.cpp

```diff
@@ -303,23 +314,49 @@ void DOS_PSP::SetFCB2(RealPt src) {
 	if (src) MEM_BlockCopy(PhysMake(seg,offsetof(sPSP,fcb2)),Real2Phys(src),16);
 }
 
-bool DOS_PSP::SetNumFiles(Bit16u fileNum) {
-	//20 minimum. clipper program.
-	if (fileNum < 20) fileNum = 20;
-	 
-	if (fileNum>20) {
+bool DOS_PSP::SetNumFiles(uint16_t file_num)
+{
+	// 20 minimum. clipper program.
+	if (file_num < 20)
+		file_num = 20;
+
+	if (file_num > 20) {
 		// Allocate needed paragraphs
-		fileNum+=2;	// Add a few more files for safety
-		Bit16u para = (fileNum/16)+((fileNum%16)>0);
-		RealPt data	= RealMake(DOS_GetMemory(para),0);
-		sSave(sPSP,file_table,data);
-		sSave(sPSP,max_files,fileNum);
-		Bit16u i;
-		for (i=0; i<20; i++)		SetFileHandle(i,(Bit8u)sGet(sPSP,files[i]));
-		for (i=20; i<fileNum; i++)	SetFileHandle(i,0xFF);
+		file_num += 2; // Add a few more files for safety
+		uint16_t para = (file_num / 16) + ((file_num % 16) > 0);
```

Comment:
The boolean-cast-to-a-1-or-0 inside the equation is pretty ugly.. 

Here's an easier to read alternative:
`i > 0 ? 1 + (i - 1) / 16 : 0`

Comparison:
https://onlinegdb.com/HJxPEPcszv

Reply:
Actually, I find it pretty readable… it's equivalent of `ceil(file_num / 16.0)` implemented in pure integer math, in branch-less style. Pretty neat. I find it a bit harder to understand your version though, so it might be subjective :)

So far I only added `const` to store the value.

Reply:
I find the code as it was written more readable… I'll keep it in (I only changed variable name here after all).

Reply:
Ok!